controllers/capabilities/backend_controller.go

```diff
@@ -74,12 +81,46 @@ func (r *BackendReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		reqLogger.V(1).Info(string(jsonData))
 	}
 
+	// Ignore deleted Backends, this can happen when foregroundDeletion is enabled
+	// https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#foreground-cascading-deletion
+	if backend.GetDeletionTimestamp() != nil && controllerutil.ContainsFinalizer(backend, backendFinalizer) {
+		// Attempt to remove backend only if backend.Status.ID is present
+		if backend.Status.ID != nil {
+			requeue, err := r.removeBackend(backend.Spec.ProviderAccountRef, *backend.Status.ID, backend.Namespace, backend.Spec.SystemName)
```

Comment:
The workflow should be:

* Remove product CR's backend references. Update product CR's if backend references were updated
* If at least one product CR was updated, requeue 
* if no product CR was updated, call 3scale api to delete the backend

Reply:
Hey, sorry, I seem to be missing the purpose of this comment. The current flow is:
- retrieve product CRs that belong to the tenant
- remove any mentions of backend in product CR, one by one where if updated - we are doing requeue.
- if there is no updates to product CR - we are removing backend 

What am I missing?

Reply:
the current implementation, returns with requeue order every time a product CR is updated. If there are N products being updated, N requeues.

What I suggest is to update all product CRs (when backend references were removed) in a single run. After that, requeue if at least one product CR was updated. If no product CR was updated, then remove backend. If there are N products being updated, 1 requeue. Besides, IMO, code is much simpler to understand (no early return in https://github.com/3scale/3scale-operator/pull/723/files#diff-e35c4c9b422525ea6dec6faafdc3acd7f0e376fd2c8a180c233eb7d0cbf19041R256)

Reply:
I get it, good point. Thanks, done!